nemo_aligner/utils/utils.py

```diff
@@ -367,8 +367,9 @@ def convert_to_amp_o2_format(state_dict):
     new_state_dict = {}
 
     for key in state_dict.keys():
-        new_key = key.replace("model.", "model.module.", 1)
-        new_state_dict[new_key] = state_dict[key]
+        if "model.module." not in key:
```

Comment:
just curious -- what caused you to need a guard like this? I guess if the weight is already O2 it'll have this issue

Reply:
I tried some old nemo models, it seems model.module. is in their key